packages/mui-base/src/utils/appendOwnerState.ts

```diff
@@ -29,7 +29,7 @@ export type AppendOwnerStateReturnType<
 /**
  * Appends the ownerState object to the props, merging with the existing one if necessary.
  *
- * @param elementType Type of the element that owns the `existingProps`. If the element is a DOM node, `ownerState` is not applied.
+ * @param elementType Type of the element that owns the `existingProps`. If the element is undefined or a DOM node, `ownerState` is not applied.
```

Comment:
nit: the DOM node case is likely much more common, so I'd put it first.

```suggestion
 * @param elementType Type of the element that owns the `existingProps`. If the element is a DOM node or undefined, `ownerState` is not applied.
```